xdsl/dialects/builtin.py

```diff
@@ -896,147 +892,6 @@ def verify(self, attr: Attribute, constraint_context: ConstraintContext) -> None
         constraint.verify(attr, constraint_context)
 
 
-@irdl_attr_definition
```

Comment:
Why move this down? What's the change here? Git isn't being very helpful

Reply:
As briefly mentioned in the PR description, though likely not explained in sufficient detail, I had to move `RankedVectorOrTensorOf` after `MemRefType`, as I changed it to now also accept `MemRefType`. As a follow-up, `DenseIntOrFPElementsAttr` also had to be moved down since it uses `RankedVectorOrTensorOf`.

As far as I remember, the only change required to `DenseIntOrFPElementsAttr` was to change the signature of the `type` argument in `from_list` from `RankedVectorOrTensorOf[AnyFloat | IntegerType | IndexType]` to:
```
RankedVectorOrTensorOf[AnyFloat | IntegerType | IndexType]
            | RankedVectorOrTensorOf[AnyFloat]
            | RankedVectorOrTensorOf[IntegerType]
            | RankedVectorOrTensorOf[IndexType]
```

Not fully clear why, but it does make pyright happy.

BTW, accepting new name suggestions for `RankedVectorOrTensorOf`.

Reply:
Makes a lot of sense, thank you